### PR TITLE
Update to eslint 3, do some small refactoring

### PR DESCRIPTION
**Pull request by lencioni**
Best reviewed commit-by-commit.

### CODE REVIEW

File: lib/environments/meteorEnvironment.js [hunk 1]
```diff
@@ -308,7 +300,8 @@ function meteorPackageNamedExports({ config }: Object): Object {
   // Try to identify the exports of all packages identified by
   // meteorPackageDependencies
   const namedExports = {};
-  for (const meteorPkg of meteorPackageDependencies({ config })) {
+
+  meteorPackageDependencies({ config }).forEach((meteorPkg: string) => {
```
Comment (lencioni) [inline lib/environments/meteorEnvironment.js]: This change was supposed to go in the previous commit, but I accidentally put it here. FYI.

**Comment by lencioni** [comment]
I'm going to merge this so I can keep working. Feel free to review anyway and I'll follow up with any changes.

File: lib/environments/meteorEnvironment.js (hunk 1 above)
Comment (trotzig) [inline lib/environments/meteorEnvironment.js]: cool.

File: lib/ImportStatements.js [hunk 2]
```diff
@@ -250,7 +250,7 @@ export default class ImportStatements {
     });
 
     if (groups.length) {
-      groups.filter((group: Array<ImportStatement>): boolean => !!group); // compact
+      groups.filter(Boolean); // compact
```
Comment (trotzig) [inline lib/ImportStatements.js]: Thanks for teaching me about this trick!

**Comment by trotzig** [comment]
Looks good to me! Thanks @lencioni!

File: lib/environments/meteorEnvironment.js [hunk 3]
```diff
@@ -276,24 +276,16 @@ function extractExportsFromMeteorPackage(
...
-      // collection.
-      declaredExports.push(declaredExport.name);
     });
 
     // TODO: If the "resources" section of the buildIsopack specifies a
     // mainModule, we need to attempt to scan it to find exports.
   });
 
-  return declaredExports;
+  return Array.from(declaredExports);
```
Comment (rhettlivingston) [inline lib/environments/meteorEnvironment.js]: In trying to use Sets more myself, I ran into the problem that Flow has a bug causing it to errantly complain about a type mismatch with the normal [...declaredExports] syntax. I'm guessing you ran into it too. I've found that an explicit cast gets around the problem, i.e. `[...(declaredExports: any)]`. A true fix is in the works in Flow.

Comment (lencioni) [inline lib/environments/meteorEnvironment.js]: Oh, good to know. I actually hadn't run into that.

Comment (trotzig) [inline lib/environments/meteorEnvironment.js]: This bug? https://github.com/facebook/flow/issues/1059#issuecomment-241267694

I'm a happy subscriber of that issue. :)

Comment (rhettlivingston) [inline lib/environments/meteorEnvironment.js]: Yep, though I've been following a PR that fixes it - facebook/flow#1668.

That syntax seems to be what the community is converging on as most common for the conversion. It's also used in most examples of how to work around the lack of a built-in union operator,,, like [this one](http://www.2ality.com/2015/01/es6-set-operations.html).

As Pascal was my first "real" language (circa 1982 after Cobol, Basic, and some assembly), I learned to appreciate sets early on, especially in language processing applications.  You'll probably see me using more of them.

**Pull request merged**